specification/advisor/resource-manager/Microsoft.Advisor/2017-04-19/swagger/advisor.json

```diff
@@ -269,65 +610,33 @@
 				"deprecated": false,
 				"x-ms-examples": {
 					"DeleteSuppression": {
-						"$ref": "./examples/DeleteSuppression.json"
-					}
-				}
-			}
-		},
-		"/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}": {
-			"get": {
-				"tags": ["GetRecommendations"],
-				"description": "Obtains details of a cached recommendation.",
-				"operationId": "Recommendations_Get",
-				"parameters": [{
-						"name": "resourceUri",
-						"in": "path",
-						"description": "The fully qualified Azure Resource Manager identifier of the resource to which the recommendation applies.",
-						"required": true,
-						"type": "string"
-					}, {
-						"name": "recommendationId",
-						"in": "path",
-						"description": "The recommendation ID.",
-						"required": true,
-						"type": "string"
-					}, {
-						"$ref": "#/parameters/apiVersionParameter"
-					}
-				],
-				"responses": {
-					"200": {
-						"description": "OK. Successfully got recommendation detail.",
-						"schema": {
-							"$ref": "#/definitions/ResourceRecommendationBase"
-						}
-					}
-				},
-				"deprecated": false,
-				"x-ms-examples": {
-					"GetRecommendationDetail": {
-						"$ref": "./examples/GetRecommendationDetail.json"
+						"$ref": "../examples/DeleteSuppression.json"
 					}
 				}
 			}
 		},
 		"/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/suppressions": {
 			"get": {
-				"tags": ["Suppressions"],
+				"tags": [
+					"Suppressions"
+				],
 				"description": "Retrieves the list of snoozed or dismissed suppressions for a subscription. The snoozed or dismissed attribute of a recommendation is referred to as a suppression.",
 				"operationId": "Suppressions_List",
-				"parameters": [{
+				"parameters": [
+					{
 						"$ref": "#/parameters/subscriptionIdParameter"
-					}, {
+					},
+					{
 						"$ref": "#/parameters/apiVersionParameter"
-					}, {
+					},
+					{
 						"name": "$top",
 						"in": "query",
 						"description": "The number of suppressions per page if a paged version of this API is being used.",
 						"required": false,
-						"type": "integer",
-						"format": "int32"
```

Comment:
Keep in mind, for some languages this will cause breaking changes to their SDK. In Go for example, there isn't any implicit conversion so [one cannot seamlessly pass an `int` to an `int32`](https://play.golang.org/p/rXqYpY8Qy7) or vice-versa.

Reply:
Should we just pick a consistent type and go with it? For example, I see there is another "integer" exposed on this same swagger spec but it is "integer" instead. I would like to either consistently name our property types/formats to "integer" or "int32" and not waffle between them.

Reply:
Good question, the SDKs that get generated at times only support the most recent API Version. For that reason, there's some value in leaving small troubles like this alone for the sake of preventing unforeseen problems. Someday, when all SDKs truly support Multiple API Versions, it will be easier to make changes, so long as they're not inside of a single API Version.